File: CoreFlows/examples/C/SinglePhase_2DLidDrivenCavity_unstructured.cxx
#include "SinglePhase.hxx"

using namespace std;

int main(int argc, char** argv)
{
	/* preprocessing: mesh and group creation */
	cout << "Loading unstructured mesh for test SinglePhase_2DLidDrivenCavity_unstructured()" << endl;
	double xinf=0.0;
	double xsup=1.0;
	double yinf=0.0;
	double ysup=1.0;
	Mesh M("resources/BoxWithMeshWithTriangularCells.med");
	double eps=1.E-6;
	M.setGroupAtPlan(xsup,0,eps,"wall");
	M.setGroupAtPlan(xinf,0,eps,"wall");
	M.setGroupAtPlan(yinf,1,eps,"wall");
	M.setGroupAtPlan(ysup,1,eps,"MovingWall");
	int spaceDim = M.getSpaceDimension();

	// physical constants
	vector<double> viscosite(1)  ;
	viscosite[0]= 0.025;

	/* set the limit field for each boundary*/
	LimitField limitWall;
	map<string, LimitField> boundaryFields;
	limitWall.bcType=Wall;
	limitWall.T = 273;
	limitWall.p = 1e5;
	limitWall.v_x = vector<double>(1,0);
	limitWall.v_y = vector<double>(1,0);
	limitWall.v_z = vector<double>(1,0);
	boundaryFields["wall"]= limitWall;

	LimitField limitMovingWall;
	limitMovingWall.bcType=Wall;
	limitMovingWall.T = 273;
	limitMovingWall.p = 1e5;
	limitMovingWall.v_x = vector<double>(1,1);
	limitMovingWall.v_y = vector<double>(1,0);
	limitMovingWall.v_z = vector<double>(1,0);
	boundaryFields["MovingWall"]= limitMovingWall;


	SinglePhase  myProblem(Liquid,around1bar300K,spaceDim);
	int nbPhase = myProblem.getNumberOfPhases();
	// Prepare for the initial condition
	int nVar = myProblem.getNumberOfVariables();
	Vector VV_Constant(nVar);
	// constant vector
	VV_Constant(0) = 1e5;
	VV_Constant(1) = 0;
	VV_Constant(2) = 0;
	VV_Constant(3) = 273;

	//Initial field creation
	cout << "Setting initial data " << endl;
	myProblem.setInitialFieldConstant(M,VV_Constant);

	//set the boundary conditions
	myProblem.setBoundaryFields(boundaryFields);

	// physical parameters
	myProblem.setViscosity(viscosite);

	// set the numerical method
	myProblem.setNumericalScheme(upwind, Implicit);

	// set the Petsc resolution
	myProblem.setLinearSolver(GMRES,ILU,true);

	// name file save
	string fileName = "2DLidDrivenCavity_unstructured";

	// parameters calculation
	unsigned MaxNbOfTimeStep = 3 ;
	int freqSave = 1;
	double cfl = 5;
	double maxTime = 5;
	double precision = 1e-8;

	myProblem.setCFL(cfl);
	myProblem.setPrecision(precision);
	myProblem.setMaxNbOfTimeStep(MaxNbOfTimeStep);
	myProblem.setTimeMax(maxTime);
	myProblem.setFreqSave(freqSave);
	myProblem.setFileName(fileName);
	myProblem.setNewtonSolver(precision,20);
	myProblem.saveVelocity();

	// evolution
	myProblem.initialize();

	bool ok = myProblem.run();
	if (ok)
		cout << "Simulation "<<fileName<<" is successful !" << endl;
	else
		cout << "Simulation "<<fileName<<"  failed ! " << endl;

	cout << "------------ End of calculation !!! -----------" << endl;
	myProblem.terminate();

	return EXIT_SUCCESS;
}

File: CoreFlows/examples/Python/SinglePhase/SinglePhase_2DLidDrivenCavity_unstructured.py
#!/usr/bin/env python
# -*-coding:utf-8 -*


import CoreFlows as cf

def SinglePhase_2DLidDrivenCavity_unstructured():
	spaceDim = 2;

	print( "Loading unstructured mesh " );
	inputfile="../resources/BoxWithMeshWithTriangularCells.med";	
 
    # set the limit field for each boundary

	fixedWallVelocityX=0;
	fixedWallVelocityY=0;
	fixedWallTemperature=273;

	movingWallVelocityX=1;
	movingWallVelocityY=0;
	movingWallTemperature=273;
	
    # physical constants

	viscosite=[0.025];

	myProblem = cf.SinglePhase(cf.Gas,cf.around1bar300K,spaceDim);
	nVar = myProblem.getNumberOfVariables();

	#Initial field creation
	print("Building initial data " ); 
	
    # Prepare for the initial condition
    
	VV_Constant = [0] * nVar

	# constant vector
	VV_Constant[0] = 1e5;
	VV_Constant[1] = 0 ;
	VV_Constant[2] = 0;
	VV_Constant[3] = 273;

    #Initial field creation
	print("Setting mesh and initial data" );
	myProblem.setInitialFieldConstant(inputfile,VV_Constant);

    # Set the boundary conditions
	myProblem.setWallBoundaryCondition("BAS", fixedWallTemperature, fixedWallVelocityX, fixedWallVelocityY);
	myProblem.setWallBoundaryCondition("GAUCHE", fixedWallTemperature, fixedWallVelocityX, fixedWallVelocityY);
	myProblem.setWallBoundaryCondition("DROITE", fixedWallTemperature, fixedWallVelocityX, fixedWallVelocityY);
	myProblem.setWallBoundaryCondition("HAUT", movingWallTemperature, movingWallVelocityX, movingWallVelocityY);

    # set physical parameters
	myProblem.setViscosity(viscosite);

    # set the numerical method
	myProblem.setNumericalScheme(cf.pressureCorrection, cf.Implicit);
	myProblem.setLinearSolver(cf.GMRES,cf.ILU,True);
   
    # name file save
	fileName = "2DLidDrivenCavityUnstructured";

    # simulation parameters
	MaxNbOfTimeStep = 3 ;
	freqSave = 1;
	cfl = 10;
	maxTime = 50;
	precision = 1e-9;

	myProblem.setCFL(cfl);
	myProblem.setPrecision(precision);
	myProblem.setMaxNbOfTimeStep(MaxNbOfTimeStep);
	myProblem.setTimeMax(maxTime);
	myProblem.setFreqSave(freqSave);
	myProblem.setFileName(fileName);
	myProblem.setNewtonSolver(precision,20);
	myProblem.saveConservativeField(True);
	if(spaceDim>1):
		myProblem.saveVelocity();
		pass

    # evolution
	myProblem.initialize();

	ok = myProblem.run();
	if (ok):
		print( "Simulation python " + fileName + " is successful !" );
		pass
	else:
		print( "Simulation python " + fileName + "  failed ! " );
		pass

	print( "------------ End of calculation !!! -----------" );

	myProblem.terminate();
	return ok

if __name__ == """__main__""":
    SinglePhase_2DLidDrivenCavity_unstructured()
